lib/AST/ASTImporter.cpp

```diff
@@ -2180,7 +2180,28 @@ Decl *ASTNodeImporter::VisitRecordDecl(RecordDecl *D) {
         if (!ToDescribed)
           return nullptr;
         D2CXX->setDescribedClassTemplate(ToDescribed);
-
+        if (!DCXX->isInjectedClassName() && D->isCompleteDefinition()) {
+          // In a record describing a template the type should be a
+          // InjectedClassNameType. Here the previously set type is updated
```

Comment:
Where is it set previously? In the ctor of CXXRecordDecl?

Reply:
The TypeForDecl is set in the CXXRecordDecl constructor, or at `Importer.getToContext().getTypeDeclType` call. The information to call `ToDescribed->getInjectedClassNameSpecialization()` is not available at this place. (This way of setting the type is got from `Sema::CheckClassTemplate`.)

Reply:
I am not sure this is a perfectly good approach. One of my concern is the FIXME below. Would the `DelayTypeCreation` parameter of the static factory help you there? Also please add the source of this method in the comment (i.e.: refer to Sema::CheckClassTemplate).